### ISSUE TITLE
Draft SC "Label gaps"

### ISSUE THREAD

**Issue by detlevhfischer** [w3c/wcag3#194]
This WCAG 2.2 proposal "Label Gaps" relates to the broader LVTF SC proposal "Proximity of Related Information: Users know about and can find related information". It takes a narrower approach, focussing on label-labelled element pairs. I hope this way it may meet the testability criterion.
This is a draft text for the proposed SC "Label Gaps":

> Label gaps
> When one element acts as a label of another element, the gap between the two is no larger than twice the width or height of the element labelled (whichever is smaller).

I.e., the simple formula is MIN(e.width;e.height)*2. This may be measured with an online ruler  (e.g. Web Developer Toolbar / Miscallaneous / Display Ruler), possibly also automatically by a script processing the absolute positions and dimensions of two elements to work out the pixel offset.
Here are two illustrations of the concept:

![name-label](https://user-images.githubusercontent.com/9989439/55013743-079a3900-4fea-11e9-9d43-b218f141283f.png)

_Alternative text:_ The first image shows a label ("First name") followed by an input field. 
The top half shows the pass condition: The label gap is 16 mm, the hight of the text field is also 16 px. Even a label gap of 32 px (twice the field height) would still meet the SC. 
The bottom half shows the FAIL condition: the horizontal gap is 33 px so it is just above the allowed maximum label gap of twice the element's width of 16 px.

----

![search-label](https://user-images.githubusercontent.com/9989439/55013761-1385fb00-4fea-11e9-9f79-245e11ba79ca.png)

_Alternative text:_ The second image shows a label ("Search") above and slightly to the left of a search input field. (This is not uncommon when a heading, search, doubles as a visual label for the field). The hight of the text field is again 16 px.  

The top half shows a pass condition: The vertical label gap is 30 px while the horizontal label gap is only 10 px. Even a label gap of 32 px (twice the field height) would still meet the SC.
The bottom half shows the FAIL condition: the vertical gap is 33 px so it is just above the allowed maximum label gap of twice the element's width of 16 px.

Obviously, the formula for measuring this is up for discussion, this is a first approximation. Possible exceptions where exceeding the distance is OK might be:

- Field is redundantly labelled by placeholder or a (conventional) icon (think: loupe)
- Left-aligned group of labels where indvidual label fails because its text is shorter
- There are guiding lines from label to labelled elements (independently, these will need meet contrast requirements of 3:1 according  to SC 1.4.11)

**Comment by adampage** [comment]
The “[Relationships detectable](https://w3c.github.io/wcag3/guidelines/#relationships-detectable)” provision in development currently says:

> [Relationships of meaning](https://w3c.github.io/wcag3/guidelines/#dfn-relationship-of-meaning) between elements are conveyed programmatically.

Must this provision be limited to programmatic relationships, or could it additionally require _visually clear_ relationships?

Related: #175

**Comment by patrickhlauke** [comment]
once you try and define explicitly how to visually represent something "clearly", you get into making arbitrary-ish graphic/visual design statements
